### PR TITLE
thing: supported boards: Add KNoT DK description

### LINKED ISSUE #16: Create description page for KNoT DK
Describe (not necessarily in that order):
 - [x] Overview
 - [x] Supported peripherals
 - [x] Pinout
 - [x] Flash instructions
 - [x] Suggested examples

### PR DESCRIPTION
**Pull request by alla3**
Closes #16

### CODE REVIEW

File: doc/thing/supported-boards/knot-dk.rst [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
++=======+================+=======+============+
+| AIN 0 | Unavailable    | P0.02 | Available  |
++-------+----------------+-------+------------+
+| AIN 1 | A0             | P0.03 | Available  |
++-------+----------------+-------+------------+
+| AIN 2 | A1             | P0.04 | Available  |
++-------+----------------+-------+------------+
+| AIN 3 | Unavailable    | P0.05 | Available  |
++-------+----------------+-------+------------+
+| AIN 4 | A3             | P0.28 | Available  |
```
Comment (jcorde) [inline doc/thing/supported-boards/knot-dk.rst]: Isn't this one A2?

File: doc/thing/supported-boards/knot-dk.rst [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
++-------+----------------+-------+------------+
+| AIN 5 | A4             | P0.29 | Available  |
```
Comment (jcorde) [inline doc/thing/supported-boards/knot-dk.rst]: A3

File: doc/thing/supported-boards/knot-dk.rst [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
++-------+----------------+-------+------------+
+| AIN 6 | A5             | P0.30 | Available  |
```
Comment (jcorde) [inline doc/thing/supported-boards/knot-dk.rst]: A4

File: doc/thing/supported-boards/knot-dk.rst [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+| LED 3 | Application | Green | P0.15 | SB7           | Available  |
++-------+-------------+-------+-------+---------------+------------+
+| LED 4 | Application | Green | P0.16 | SB8           | Available  |
++-------+-------------+-------+-------+---------------+------------+
+
+Analog Input
+''''''''''''
+
++-------+----------------+-------+------------+
+| Name  | Arduino Naming | Pin   | Permission |
```
Comment (jcorde) [inline doc/thing/supported-boards/knot-dk.rst]: Please, see [this](https://infocenter.nordicsemi.com/topic/ug_nrf52840_dk/UG/nrf52840_DK/hw_analog_pins.html) page

File: doc/thing/supported-boards/knot-dk.rst [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+
+-------------------------------------------------------------------------------
+
+Flash instructions
+------------------
+
+To flash the KNoT DK connect a USB cable between KNoT DK and a PC using the J2
+connector and turn the power switch on. Make sure the SW9 switch is on VDD and the
+SW6 switch is on default position. Use nRF Connect or KNoT CLI to flash the
+image on your board.
```
Comment (jcorde) [inline doc/thing/supported-boards/knot-dk.rst]: Please, emphasize the options as in _VDD_, _DEFAULT_, _SW6_, ...

Comment (jcorde) [reply doc/thing/supported-boards/knot-dk.rst]: Please, link KNoT CLI to the target section thing/thing-cli.html#flash-board-when-done

File: doc/thing/supported-boards/knot-dk.rst [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+.. figure:: ../../../_static/dk_flash.png
+   :scale: 50 %
+   :alt: DK - nRF52840 (PCA10056)
+   :align: center
+
+   Switches position when flashing KNoT DK
```
Comment (jcorde) [inline doc/thing/supported-boards/knot-dk.rst]: I would use `Position of switches to flash KNoT DK`. That would avoid the user to mistake it with "The position is switched when flashing KNoT DK".

File: doc/thing/supported-boards/knot-dk.rst [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+
+Buttons
+'''''''
+
++-------+------------+---------------+---------------+------------+
+| Part  | Name       | Description   | Pin           | Permission |
++=======+============+===============+===============+============+
+| SW1   | Button 1   | Factory Reset | P0.11 / P1.07 | Reserved   |
++-------+------------+---------------+---------------+------------+
+| SW2   | Button 2   | Application   | P0.12 / P1.07 | Available  |
```
[the remark this answers is not in the document]
Comment (alla3) [reply doc/thing/supported-boards/knot-dk.rst]: The P1.07 is used as an alternative pin for Button 1. It was a typo on Button 2, in this case, the correct pin would be the P1.08. Thank you.

File: doc/thing/supported-boards/knot-dk.rst [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+   - RTC
+   - Segger RTT
+   - SPI
+   - UART
+   - USB
+   - WDT
+
+-------------------------------------------------------------------------------
+
+Connections and IOs
```
[the remark this answers is not in the document]
Comment (alla3) [reply doc/thing/supported-boards/knot-dk.rst]: I agree. This information is relevant but it is necessary to perform more test cases using these device interfaces before adding it on the doc. Maybe you could create an issue for it.

**Pull request merged**